lib/betagouvbot/notification_request.rb

```diff
@@ -15,11 +15,13 @@ def call(members, date)
       private
 
       def warnings(members, date)
-        NotificationSchedule.(members, horizons, date)
+        NotificationSchedule.(members, end_dates(date))
       end
 
-      def horizons
-        NotificationRule.horizons
+      def end_dates(date)
```

Comment:
Thing is, I don't want to add external dependencies to rules and schedule. You may have already noticed the pattern: `NotificationRequest` orchestrates both, and neither `NotificationRule` nor `NotificationSchedule` depend on others.